plugins/inputs/redfish/redfish.go

```diff
@@ -0,0 +1,312 @@
+package redfish
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/internal/tls"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"io/ioutil"
+	"math"
+	"net/http"
+)
+
+type Cpu struct {
+	Name           string    `json:"Name"`
+	ReadingCelsius int64     `json:"ReadingCelsius"`
+	Status         CpuStatus `json:"Status"`
+}
+type Payload struct {
+	Temperatures  []*Cpu   `json:",omitempty"`
```

Comment:
might want to use slices of structs instead of slices of pointers to structs. This way you don't need to worry about the nil pointer case